src/apps/deskflow-gui/MainWindow.cpp

```diff
@@ -1065,6 +1072,48 @@ void MainWindow::showAndActivate()
   m_actionMinimize->setVisible(true);
 }
 
+void MainWindow::showHostNameEditor()
+{
+  ui->lineEditName->show();
+  ui->lblComputerName->hide();
+  ui->btnEditName->hide();
+  ui->lineEditName->setFocus();
+}
+
+void MainWindow::setHostName()
+{
+  ui->lineEditName->hide();
+  ui->lblComputerName->show();
+  ui->btnEditName->show();
+
+  QString text = ui->lineEditName->text();
+  bool existingScreen = serverConfig().screenExists(text) && text != m_appConfig.screenName();
+
+  if (!ui->lineEditName->hasAcceptableInput() || text.isEmpty() || existingScreen) {
+    blockSignals(true);
+    ui->lineEditName->setText(m_appConfig.screenName());
+    blockSignals(false);
+
+    const auto title = tr("Invalid Screen Name");
+    QString body;
+    if (existingScreen) {
+      body = tr("Screen name already exists");
+    } else {
+      body = tr("The name you have choosen is invalid.\n\n"
+                "Valid Names:\n"
+                "• Use letters and numbers\n"
+                "• May also use _ or - \n"
+                "• Are between 1 and 255 characters");
```

Comment:
Damn, just spotted a couple of typos:
- "choosen" -> "chosen"
- "Valid Names" -> "Valid names"

```suggestion
      body = tr("The name you have chosen is invalid.\n\n"
                "Valid names:\n"
                "• Use letters and numbers\n"
                "• May also use _ or -\n"
                "• Are between 1 and 255 characters");
```